marketplace/bynder-assets/src/index.js

```diff
@@ -1,24 +1,96 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
+import { setup } from 'shared-dam-app';
 
-import { init, locations } from 'contentful-ui-extensions-sdk';
+import logo from './logo.svg';
+import descriptor from '../extension.json';
 
-import '@contentful/forma-36-react-components/dist/styles.css';
-import '@contentful/forma-36-fcss/dist/styles.css';
-import './index.css';
+const BYNDER_SDK_URL =
+  'https://d8ejoa1fys2rk.cloudfront.net/modules/compactview/includes/js/client-1.5.0.min.js';
 
-import BynderDialog from './BynderDialog';
-import Field from './Field';
-import AppConfig from './AppConfig';
+const CTA = 'Select or upload a file on Bynder';
 
-init(sdk => {
-  const root = document.getElementById('root');
+function makeThumbnail(resource) {
+  const thumbnail = resource.thumbnails && resource.thumbnails.webimage;
+  const url = typeof thumbnail === 'string' ? thumbnail : undefined;
+  const alt = [resource.id, ...(resource.tags || [])].join(', ');
 
-  if (sdk.location.is(locations.LOCATION_DIALOG)) {
-    ReactDOM.render(<BynderDialog sdk={sdk} />, root);
-  } else if (sdk.location.is(locations.LOCATION_ENTRY_FIELD)) {
-    ReactDOM.render(<Field sdk={sdk} />, root);
-  } else if (sdk.location.is(locations.LOCATION_APP)) {
-    ReactDOM.render(<AppConfig sdk={sdk} />, root);
+  return [url, alt];
+}
+
+function prepareBynderHTML(bynderURL) {
+  return `
+    <div class="dialog-container">
+      <div
+        id="bynder-compactview"
+        data-assetTypes="image"
+        data-autoload="true"
+        data-button="Load media from bynder.com"
+        data-collections="true"
+        data-folder="bynder-compactview"
+        data-fullScreen="true"
+        data-header="false"
+        data-language="en_US"
+        data-mode="multi"
+        data-zindex="300"
+        data-defaultEnvironment="${bynderURL}"
+      />
+    </div>
+  `;
+}
+
+function renderDialog(sdk) {
+  const config = sdk.parameters.invocation;
+
+  const container = document.createElement('div');
+  container.innerHTML = prepareBynderHTML(config.bynderURL);
+  document.body.appendChild(container);
+
+  const script = document.createElement('script');
+  script.src = BYNDER_SDK_URL;
+  script.async = true;
+  document.body.appendChild(script);
+
+  sdk.window.startAutoResizer();
+
+  document.addEventListener('BynderAddMedia', e => {
+    sdk.close(Array.isArray(e.detail) ? e.detail : []);
+  });
+}
+
+async function openDialog(sdk, _currentValue, config) {
+  const result = await sdk.dialogs.openExtension({
+    position: 'center',
+    title: CTA,
+    shouldCloseOnOverlayClick: true,
+    shouldCloseOnEscapePress: true,
+    parameters: { ...config },
+    width: 1400
+  });
+
+  return Array.isArray(result) ? result : [];
+}
+
+function isDisabled() {
+  return false;
+}
```

Comment:
Is this function even needed?

Reply:
yes, `setup` function from `shared-dam-app` expects interface expressed as `Integration` (see type definitions). There is nothing specific to the Bynder app (as oppesed to Cloudinary with its asset limit) that could disable editing so we always return `false`

Reply:
Ok thanks for explaining.